src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.ProjectState.cs

```diff
@@ -199,12 +200,17 @@ public async ValueTask SaveToInMemoryStorageAsync(Project project, DiagnosticAna
 
                 RemoveInMemoryCache(_lastResult);
 
+                using var _ = PooledHashSet<DocumentId>.GetInstance(out var documentIdsToProcess);
+                documentIdsToProcess.AddRange(_lastResult.DocumentIdsOrEmpty);
+                documentIdsToProcess.AddRange(result.DocumentIdsOrEmpty);
```

Comment:
This fixes #69763 for FSA issue with the same scenario. I have also updated the added integration test for the same. I verified that the scenario in #69763 is fixed after this product change.